### PR TITLE
planner: support JSON format explain

### PR DESCRIPTION
**Pull request by fzzf678**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.3 (TiDB 6.3 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):
https://github.com/pingcap/tidb/pull/39253
### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: sql-statements/sql-statement-explain.md [hunk 1]
```diff
@@ -185,12 +185,13 @@ EXPLAIN DELETE FROM t1 WHERE c1=3;
 
 在 `EXPLAIN` 中使用 `FORMAT = "xxx"` 语法可以指定输出的内容和格式。
```
Comment (ran-huang) [inline sql-statements/sql-statement-explain.md:186]: 建议把这里的几种格式示例写成 tab 形式， @fzzf678 你觉得呢？可以的话我先这样改
image

**Comment by fzzf678** [comment]
@ran-huang  some line changed, PTAL  again.

File: sql-statements/sql-statement-explain.md [hunk 2]
```diff
@@ -309,16 +309,15 @@ EXPLAIN FORMAT = "json" SELECT id FROM t WHERE a = 1;
...
 
-在输出结果中，`id`、`estRows`、
-`task`、`access object`、`operator info` 含义与默认格式相同。`subOperators` 是存放子节点的数组，子节点的字段及含义与父节点均相同。若某字段缺失则说明该字段为空。
+在输出结果中，`id`、`estRows`、`task`、`access object`、`operator info` 含义与默认格式相同。`subOperators` 是存放子节点的数组，子节点的字段及含义与父节点均相同。若某字段缺失则说明该字段为空。
 
 ## MySQL 兼容性
 
 * `EXPLAIN` 的格式和 TiDB 中潜在的执行计划都与 MySQL 有很大不同。
-* TiDB 不支持 `FORMAT=TREE` 选项。
+* TiDB 不支持 `FORMAT=JSON` 或 `FORMAT=TREE` 选项。
```
Comment (ran-huang) [inline sql-statements/sql-statement-explain.md:320]: 这里要说明一下吗，如果需要输出为 JSON 请使用 tidb_json

**Review by ran-huang — COMMENTED** [review]
rest LGTM

File: sql-statements/sql-statement-explain.md (hunk 1 above)
Comment (ran-huang) [reply sql-statements/sql-statement-explain.md]: done

File: sql-statements/sql-statement-explain.md (hunk 2 above)
Comment (fzzf678) [reply sql-statements/sql-statement-explain.md]: tidb_json 和Mysql的 JSON 输出字段还不一样 ，`FORMAT=JSON` 打算后面给兼容MySQL时候用

File: sql-statements/sql-statement-explain.md [hunk 3]
```diff
@@ -183,14 +183,19 @@ EXPLAIN DELETE FROM t1 WHERE c1=3;
...
 
-| FORMAT | 作用 |
-| ------ | ------ |
-| 未指定  | 同 row |
-| `row`    | `EXPLAIN` 语句将以表格格式输出结果。更多信息，可参阅 [TiDB 执行计划概览](/explain-overview.md) |
-| `brief`  | `EXPLAIN` 语句输出结果中的算子 ID 将被简化，较之未指定 `FORMAT` 时输出结果的算子 ID 更为简化 |
-| `dot`    | `EXPLAIN` 语句将输出 dot 格式的执行计划，可以通过 `dot` 程序（在 `graphviz` 包中）生成 PNG 文件 |
+| FORMAT      | 作用                                                                |
+|-------------|-------------------------------------------------------------------|
+| 未指定         | 同 row                                                             |
```
Comment (qiancai) [inline sql-statements/sql-statement-explain.md:190]:
```suggestion
| 未指定         | 未指定 `FORMAT` 时，默认输出格式为 `row`                                                             |
```

File: sql-statements/sql-statement-explain.md [hunk 4]
```diff
@@ -183,14 +183,19 @@ EXPLAIN DELETE FROM t1 WHERE c1=3;
 4 rows in set (0.01 sec)
 ```
 
-在 `EXPLAIN` 中使用 `FORMAT = "xxx"` 语法可以指定输出的内容和格式。
+在 `EXPLAIN` 中使用 `FORMAT = "xxx"` 语法可以指定输出的内容和格式。目前支持的输出格式如下：
```
Comment (qiancai) [inline sql-statements/sql-statement-explain.md:186]:
```suggestion
在 `EXPLAIN` 语句中，你可以使用 `FORMAT = "xxx"` 语法指定输出内容的格式。目前支持的输出格式如下：
```

File: sql-statements/sql-statement-explain.md [hunk 5]
```diff
@@ -270,6 +278,57 @@ The `xx.dot` is the result returned by the above statement.
 
 ![Explain Dot](/media/explain_dot.png)
 
+</div>
+
+<div label="JSON">
+
+在 `EXPLAIN` 指定 `FORMAT = "tidb_json"` 时，示例如下：
```
Comment (qiancai) [inline sql-statements/sql-statement-explain.md:285]:
```suggestion
在 `EXPLAIN` 中指定 `FORMAT = "tidb_json"` 时，示例如下：
```

File: sql-statements/sql-statement-explain.md [hunk 6]
```diff
@@ -270,6 +278,57 @@ The `xx.dot` is the result returned by the above statement.
...
+            }
+        ]
+    }
+]
+ |
++------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+1 row in set (0.01 sec)
+```
+
+在输出结果中，`id`、`estRows`、`task`、`access object`、`operator info` 含义与默认格式相同。`subOperators` 是存放子节点的数组，子节点的字段及含义与父节点均相同。若某字段缺失则说明该字段为空。
```
Comment (qiancai) [inline sql-statements/sql-statement-explain.md:326]:
```suggestion
在输出结果中，`id`、`estRows`、`taskType`、`accessObject`、`operatorInfo` 与默认格式输出的列含义相同。`subOperators` 是存放子节点的数组，子节点的字段及含义与父节点均相同。若某字段缺失，则说明该字段为空。
```

File: sql-statements/sql-statement-explain.md [hunk 7]
```diff
@@ -270,6 +278,57 @@ The `xx.dot` is the result returned by the above statement.
...
+
+</div>
+
+</SimpleTab>
+
 ## MySQL 兼容性
 
 * `EXPLAIN` 的格式和 TiDB 中潜在的执行计划都与 MySQL 有很大不同。
```
Comment (qiancai) [inline sql-statements/sql-statement-explain.md:334]: L355 提到了 TiDB 不支持 FORMAT=JSON，是不是需要改下 L355

Comment (ran-huang) [reply sql-statements/sql-statement-explain.md:334]: 这个讨论过了，tidb_json 和 json 不一样：https://github.com/pingcap/docs-cn/pull/12102#discussion_r1036634987

**Review by ran-huang — APPROVED** [review]

**Review by qiancai — APPROVED** [review]
LGTM

File: sql-statements/sql-statement-explain.md [hunk 8]
```diff
@@ -270,10 +278,62 @@ The `xx.dot` is the result returned by the above statement.
...
 * TiDB 不支持 `FORMAT=JSON` 或 `FORMAT=TREE` 选项。
+* TiDB 支持的 `FORMAT=TIDB_JSON` 是对当前默认 `EXPLAIN` 格式的 JSON 编码，与 MySQL 的 `FORMAT=JSON` 结果的格式、字段信息都不同。
```
Comment (qiancai) [inline sql-statements/sql-statement-explain.md:336]:
```suggestion
* TiDB 支持的 `FORMAT=tidb_json` 是对当前默认 `EXPLAIN` 格式的 JSON 编码，与 MySQL 的 `FORMAT=JSON` 结果的格式、字段信息都不同。
```

**Review by qiancai — APPROVED** [review]
LGTM

**Pull request merged**